src/index.js

```diff
@@ -28,53 +31,69 @@ addInput.addEventListener('keypress', (event) => {
   }
 });
 
+clearAll.addEventListener('click', async () => {
+  const removeCompleted = arr.filter((arr) => arr.completed === false);
+  arr = removeCompleted;
+  defaultExport();
+});
+
 export function component() {
-  arr.forEach((card, n) => {
-    const element = document.createElement('li');
-    element.setAttribute('id', `i${card.index}`);
-    element.innerHTML += `
-      <input type="checkbox" name="task${card.index}">
+  if (Array.isArray(arr) && arr.length > 0) {
+    arr.forEach(async (card, n) => {
+      const element = document.createElement('li');
+      element.setAttribute('id', `i${card.index}`);
+      element.innerHTML += `
+      <input id="check${n}" type="checkbox" name="task${card.index}">
       <label for="task${card.index}">${card.description}</label>
       <textarea  id="task${n}" name="inputT" class="inputT">${card.description}</textarea>
       <button type="button" id="removeTask">
       <span id='delete-${n}' class="material-symbols-outlined  delete-ic">delete</span>
       <span class="material-symbols-outlined ic-vDots">more_vert</span>
       </button>
     `;
-    element.addEventListener('mouseenter', () => {
-      const deleteIC = document.getElementById(`delete-${n}`);
-      if (deleteIC) {
-        deleteIC.style.display = 'block';
-      }
-      element.querySelector('.ic-vDots').style.display = 'none';
-      element.style.backgroundColor = 'yellow';
-    });
-    element.addEventListener('mouseleave', () => {
-      const deleteIC = document.getElementById(`delete-${n}`);
-      if (deleteIC) {
-        deleteIC.style.display = 'none';
-      }
-      element.querySelector('.ic-vDots').style.display = 'block';
-      element.style.backgroundColor = 'white';
-    });
-    element.querySelector('#removeTask').addEventListener('click', async () => {
-      arrayIndex = `${card.index}` - 1;
-      const { deletingTask } = await import('./interactions.js');
-      deletingTask();
-      element.querySelector('.ic-vDots').style.display = 'none';
-    });
-    const z = element.querySelector('.inputT');
-    z.addEventListener('keypress', async (event) => {
-      if (event.key === 'Enter') {
-        event.preventDefault();
-        arr[n].description = z.value;
-        saveToLocalStorage();
-        removeAllTasks(container);
-        component();
-      }
+      element
+        .querySelector(`#check${n}`)
+        .addEventListener('click', async () => {
+          arr[n].completed = element.querySelector(`#check${n}`).checked;
+        });
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#:~:text=Add%20event%20listener%20to%20the%20checkbox), Kindly add `change` event listener on the checkbox to change the status of the task to `completed: true` when checked and  `completed: false` when unchecked and also save this change to the local storage.